Source/EmbeddedBoundary/WarpXInitEB.cpp

```diff
@@ -110,27 +112,24 @@ WarpX::ComputeEdgeLengths () {
 
     auto const &flags = eb_fact.getMultiEBCellFlagFab();
     auto const &edge_centroid = eb_fact.getEdgeCent();
-    for (amrex::MFIter mfi(flags); mfi.isValid(); ++mfi){
-        amrex::Box const &box = mfi.validbox();
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim){
+        for (amrex::MFIter mfi(flags); mfi.isValid(); ++mfi){
+        amrex::Box const &box = mfi.tilebox(m_edge_lengths[maxLevel()][idim]->ixType().toIntVect());
         amrex::FabType fab_type = flags[mfi].getType(box);
-        for (int idim = 0; idim < AMREX_SPACEDIM; ++idim){
             auto const &edge_lengths_dim = m_edge_lengths[maxLevel()][idim]->array(mfi);
             if (fab_type == amrex::FabType::regular) {
                 // every cell in box is all regular
-                amrex::LoopOnCpu(amrex::convert(box, amrex::Box(edge_lengths_dim).ixType()),
-                                 [=](int i, int j, int k) {
+                amrex::ParallelFor(box, [=](int i, int j, int k) {
```

Comment:
```suggestion
                amrex::ParallelFor(box, [=] AMREX_GPU_DEVICE (int i, int j, int k) {
```
And similar changes in other places.